src/util/datablock/oo_datablock.c

```diff
@@ -0,0 +1,26 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "oo_datablock.h"
+#include "../arr.h"
+
+inline void *DataBlock_AllocateItemOutOfOrder(DataBlock *dataBlock, uint64_t idx) {
+	// Check if idx<=data block's current capacity. If needed, allocate additional blocks.
+	DataBlock_Accommodate(dataBlock, idx);
+	DataBlockItemHeader *item_header = DataBlock_GetItemHeader(dataBlock, idx);
+	MARK_HEADER_AS_NOT_DELETED(item_header);
+	dataBlock->itemCount++;
+	return ITEM_DATA(item_header);
+}
+
+inline void DataBlock_MarkAsDeletedOutOfOrder(DataBlock *dataBlock, uint64_t idx) {
```

Comment:
Why did you decide to mark this function as `inline`?

Reply:
`inline` is just a compiler recommendation, the body is 4-5 lines. seems legit to me.